### DISCUSSION TITLE
Request timeout / no network connection?

### DISCUSSION THREAD

**Discussion by getnashty** [urql-graphql/urql#3373] [Q&A] [answered]
Hi all, I'm trying to detect when a request has timed out, or when there has been a network error, so I can navigate the user to offline mode.. I'm using react native like so:

```
        auth,
        fetchExchange,
        errorExchange({
            onError: (error) => {
                if (error.message.includes('Network') || error.networkError) {
                    navigation.dispatch(CommonActions.reset({ index: 0, routes: [{ name: 'Offline' }] }));
                }
            },
        }),
```

When I turn off the internet to test, this never seems to fire.. the loading for the request never ends.. is there a better approach to this sort of thing?

**Comment by kitten** [comment, marked as answer]
There's some more links with more details here: https://formidable.com/open-source/urql/docs/architecture/#the-exchanges

but the gist of it is that the order of exchanges matters. Your operation enters the list of exchanges like a request would do in HTTP Middleware. Results are then passed backwards, in reverse order.

This means that no result will actually be received by the `errorExchange` you added in your snippet since it's after the `fetchExchange` rather than in front of it

**Reply by getnashty** [reply]
Thank you @kitten - that fixed it for me 🙏
